### ISSUE TITLE
Project roadmap/vision/financing

### ISSUE THREAD

**Issue by jackturnbull** [spider-gazelle/spider-gazelle#7]
Wording for this sort of thing can be a little tricky so take it in good faith 😄 

I've been dipping my toes in and out of crystal for a couple of years and not built anything substantial in it and I'm now getting to the point where I'm ready to throw a real project its way. Settling on libraries and frameworks that are going to continue to be supported and see development is probably the hardest bit of Crystal right now, the language and coding can almost look after itself.

So with that being said I was hoping to tease a few things out of you guys, if you don't mind. In particular your longer term support strategy and where you see spider-gazelle going. By support, I possibly mean a number of things:

* **Lucky Framework**; Paul C. Smith has previously said that he gets (something like) half a day a week to work on side-projects funded by thoughtbot. Not the full backing of a major company but I believe him and one other tackle a few issues here and there a week. _Not guaranteeing this to be accurate, just light observations._
* **Amber/Kemal**; Safety in numbers feels like it can apply, I see PRs and issues being raised by quite diverse sections of the Crystal community which is pretty encouraging on the whole. I suspect there are a few mid-size production applications out there in the wild as well, by this point anyway.

So without sounding rude - who are you guys? And what sorts of projects are you using spider-gazelle for? How long do you intend to support those projects?

And to take it a step further, what sort of direction do you plan on taking spider-gazelle? Is it pretty close to 'fleshed out' or are there more radical ideas you're considering?

**Comment by stakach** [comment]
Hi @jackturnbull all good questions.
Spider-Gazelle (SG) is sponsored by https://www.acaprojects.com/

We have a single project team working on Crystal projects at the moment (it was just me up until recently) and we have started to migrate all our products over to crystal.

Spider-Gazelle is more or less feature complete. We are still building out Active Model at the moment (although Amber et al also has good ORMs that work with SG). The ORMs being built at the moment are targeting Redis and RethinkDB.

There is a plan to implement a GraphQL API which might involve adding helpers to SG.

Support at minimum will be the lifetime of a building / workplace so no less than 5 years but realistically it should be longer - which is the plan and the initial release date of the crystallised version of our primary product is in March.

**Comment by jackturnbull** [comment]
Thanks, this is good to know! I'm a big fan of the documentation and I like the current simplicity of it all. There's certain things I'll need to add around it but in some ways I prefer that too. I'll be sure to keep an eye on it as you approach your release candidate dates.

I'll try and get one more out of you, if you'll allow it: _what do you see as being the primary goals for spider-gazelle? What problems do you want to solve that you couldn't manage with other frameworks?_ To answer it with what I know about other web frameworks out there; Lucky might say that they wish to remove all possible type errors at the compilation stage, Amber possibly that they wish to make the developer experience and migration from Rails as easy as possible. Would you want it to do everything out of the box, or keep it as light and minimal as possible, for example?

Essentially is there a particular itch you wish to scratch with this project?

**Comment by stakach** [comment]
Well we started the project when after looking at https://github.com/the-benchmarker/web-frameworks where router.cr was way faster than all the other frameworks (they've since updated how the tests are run) and still states "Kemal or other web frameworks written in Crystal are awesome, but too fat for some purpose" - on top of that Amber seemed complex, lucky too verbose with raze and kemel too simplistic.
Amber's the framework we probably would have used if we didn't write our own.

We more or less were aiming for "Crystal on Rails" where we could improve some things that confuse people when they start working on our projects.
For example controllers defining routes seemed more logical than having a seperate configuration file.
Simplified testing, more control over application launching and super easy to deploy.

Basically we write a lot of micro services that have a REST interface however that's secondary to what the application actually does. We want to empower our developers, keeping things simple and easy to use without sacrificing performance.

**Comment by jackturnbull** [comment]
Thanks for your responses here, it definitely helps put it all into context. I've been in a somewhat similar boat that I'm after something familar but weighing up whether to go for a more feature-complete framework that I'd be comfortable in, vs something lighter that I'm more unfamilar with has been a tough call also.

I don't have much else to add at this point, so you can feel free to close the issue if you wish? I'd be happy leaving it open too if you prefer it for discussion.
